### PR TITLE
Update documentation and action to _allow_ github tokenless

### PR DESCRIPTION
**Pull request by Ross-ForAllSecure**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,8 +46,10 @@ Want to try it? [Sign up for free](http://mayhem4api.forallsecure.com/signup) to
 3. Add the following text to the file and tweak it for your codebase
 
 ```yaml
-name: Example workflow for Mayhem for API
+name: Mayhem for API
 on: push
+env:
```
Comment (swarkentin) [inline README.md:51]: Does this need to be explicitly set? 

For instance - can we just access this like: https://github.com/ForAllSecure/mcode-action/blob/main/src/main.ts#L37 ?

Comment (Ross-ForAllSecure) [reply README.md:51]: Everything else in the set of github credentials set is getting pulled in from the environment.  $GITHUB_RUN_ID and $GITHUB_REPOSITORY.

Comment (swarkentin) [reply README.md:51]: More specifically, if this block is absent:

```
env:
  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Will the action still work with GitHub tokenless?

**Comment by swarkentin** [comment]
lgtm! Will need to post a new version once this goes in.

**Pull request merged**